provider/azure/internal/ad/client.go

```diff
@@ -69,8 +77,30 @@ func WithOdataErrorUnlessStatusCode(codes ...int) autorest.RespondDecorator {
 
 				// Copy and replace the Body in case it does not contain an error object.
 				// This will leave the Body available to the caller.
-				b, decodeErr := autorest.CopyAndDecode(autorest.EncodedAsJSON, resp.Body, &oe)
+				var b bytes.Buffer
+				if _, err := io.Copy(&b, resp.Body); err != nil {
+					return err
+				}
+				var decodeErr error
```

Comment:
As for the simplification: I'm intentionally preserving the behaviour of leaving the body as-is for the caller. But I think you're right, we should just chuck out the BOM and pretend it was never there. It's no use to anyone.